trans/tests/TransSpec.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+
+module Main where
+
+import Data.Decimal
+import Test.Hspec
+import Test.QuickCheck
+
+import qualified Pact.Trans.Musl as Musl
+import qualified Pact.Trans.Dec as Dec
+import qualified Pact.Trans.Dbl as Dbl
+import Pact.Trans.Types
+
+instance Arbitrary i => Arbitrary (DecimalRaw i) where
+  arbitrary = Decimal <$> arbitrary <*> arbitrary
+
+-- WARNING: These tests are currently a no-op, because there has been no way
+-- discovered yet to reach agreement between either native doubles and MPFR,
+-- or Musl and MPFR.
+
+main :: IO ()
+main = hspec $ do
+  describe "TransSpec" $ do
+    floatingPointTestsArity1 "exp"  Dbl.dbl_exp  Musl.musl_exp  Dec.dec_exp
+    floatingPointTestsArity1 "ln"   Dbl.dbl_ln   Musl.musl_ln   Dec.dec_ln
+    floatingPointTestsArity2 "log"  Dbl.dbl_log  Musl.musl_log  Dec.dec_log
+    floatingPointTestsArity2 "pow"  Dbl.dbl_pow  Musl.musl_pow  Dec.dec_pow
+    floatingPointTestsArity1 "sqrt" Dbl.dbl_sqrt Musl.musl_sqrt Dec.dec_sqrt
+
+dec2F :: Decimal -> Double
+dec2F = fromRational . toRational
+
+f2Dec :: Double -> Decimal
+f2Dec = fromRational . toRational
+
+{-
+From the MPFR 4.1.0 documentation:
+
+"...with a precision of 53 bits and in any of the four standard rounding
+modes, MPFR is able to exactly reproduce all computations with double-
+precision machine floating-point numbers (e.g., double type in C, with a C
+implementation that rigorously follows Annex F of the ISO C99 standard and
+FP_CONTRACT pragma set to OFF) on the four arithmetic operations and the
+square root, except the default exponent range is much wider and subnormal
+numbers are not implemented (but can be emulated)."
+-}
+
+floatingPointTestsArity1
+  :: String
+  -> (Double -> TransResult Double)
+  -> (Double -> TransResult Double)
+  -> (Decimal -> TransResult Decimal)
+  -> Spec
+floatingPointTestsArity1 lbl f_double f_musl f_mpfr =
+  it ("floatingPointTestsArity1::" ++ lbl) $
+    property $ withMaxSuccess 1_000_000  $ \x -> do
```

Comment:
With this setting, we exceed GH actions timelimit

Reply:
These tests are no longer meaningful, since I can't find any double regime that exactly matches any other. I've left them in the code in case I want it in the future, but once this is merged we should delete it and leave it to the Git history.